### PR TITLE
Initial version of N4JS design document

### PR DESCRIPTION
**Pull request by jpilgrim**
#1173 

With documentation taken from my presentations (at EclipseCon and XtextCon), overview of components (by features), and some content from the readme.

### CODE REVIEW

File: docs/org.eclipse.n4js.design/build.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+bin.includes = META-INF/
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.design/build.properties]: File `docs/org.eclipse.n4js.design/.project` is missing, so new project cannot be imported into Eclipse workspace. Same for settings folder (best copy that from plugin containing the spec).

File: docs/org.eclipse.n4js.design/plugin.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
+pluginName = org.eclipse.n4js.spec
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.design/plugin.properties]: should be `.design` at the end

File: docs/org.eclipse.n4js.design/build.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.design/build.sh]: a `.gitignore` file should be added, similar to the one in the spec plugin

File: docs/org.eclipse.n4js.design/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+Bundle-SymbolicName: org.eclipse.n4js.spec;singleton:=true
+Bundle-Version: 0.0.1.qualifier
+Bundle-Vendor: %providerName
+Require-Bundle: org.eclipse.help,
+ org.eclipse.swt,
+ org.eclipse.jface,
+ org.eclipse.ui,
+ org.eclipse.n4js.ui;bundle-version="0.0.1"
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Automatic-Module-Name: org.eclipse.n4js.spec
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.design/META-INF/MANIFEST.MF]: value should be `org.eclipse.n4js.design`

File: docs/org.eclipse.n4js.design/chapters/01_introduction/introduction.adoc [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+We reuse the notation specified in cite:[N4JSSpec].
+
+
+
+[[sec:IDE_Overview]]
+== IDE Components
+
+The N4JS and N4JSIDE components are organized via features. The following features with included plugins are defined (the common prefix "org.eclipse.n4js" is omitted):
+
+[horizontal]
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.design/chapters/01_introduction/introduction.adoc]: The following large table isn't very readable without any borders.

File: docs/org.eclipse.n4js.doc/web-html/articles/index.html [hunk 6]
```diff
@@ -0,0 +1,211 @@
+<!DOCTYPE html>
```
Comment (mor-n4) [inline docs/org.eclipse.n4js.doc/web-html/articles/index.html]: You added a new folder `docs/org.eclipse.n4js.doc/web-html` with many new files. I guess this was by accident.
If this is the case, please remove!

(I noticed that there is a gitignore entry for `web-html`, but only in subfolder `www` of plugin `org.eclipse.n4js.doc`. Maybe there's a glitch in the script for generating the web site?)

File: docs/org.eclipse.n4js.design/build.properties (hunk 1 above)
Comment (jpilgrim) [reply docs/org.eclipse.n4js.design/build.properties]: added along with .settings and .classpath from spec

File: docs/org.eclipse.n4js.design/build.sh (hunk 3 above)
Comment (jpilgrim) [reply docs/org.eclipse.n4js.design/build.sh]: added

**Pull request merged**